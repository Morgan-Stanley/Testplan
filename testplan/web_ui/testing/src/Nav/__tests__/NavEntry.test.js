import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";
import {Badge} from 'reactstrap';

import NavEntry from '../NavEntry';

function defaultProps() {
  return {
    name: 'entry name',
    status: 'passed',
    type: 'testplan',
    caseCountPassed: 0,
    caseCountFailed: 0,
  };
}

describe('NavEntry', () => {
  const props = defaultProps();

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('shallow renders the correct HTML structure', () => {
    const navEntry = shallow(
        <NavEntry {...props} />
    );
    expect(navEntry).toMatchSnapshot();
  });

  it('when prop status="failed" name div and Badge have correct styles',
     () => {
    const failProps = {...props, status: 'failed'}
    const navEntry = shallow(
      <NavEntry {...failProps} />
    );

    const name = navEntry.children().first();
    expect(name).toHaveLength(1);
    const expectedName = defaultProps().name;
    expect(name.text()).toEqual(expectedName);
    expect(name.props().className).toMatch(/failed/);

    const badge = navEntry.find(Badge);
    expect(badge).toHaveLength(1);
    expect(badge.props().className).toMatch(/failedBadge/);
  });
});
