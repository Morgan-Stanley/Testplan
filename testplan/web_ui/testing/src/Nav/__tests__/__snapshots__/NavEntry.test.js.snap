// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavEntry shallow renders the correct HTML structure 1`] = `
<div
  className="d-flex justify-content-between"
>
  <div
    className="entryName_udjila-o_O-passed_1kh4m98"
    title="entry name"
  >
    entry name
  </div>
  <div
    className="entryIcons_dx1noq"
  >
    <i
      className="entryIcon_ull2p3"
      title="passed/failed testcases"
    >
      <span
        className="passed_1kh4m98"
      >
        0
      </span>
      /
      <span
        className="failed_11es5k7"
      >
        0
      </span>
    </i>
    <Badge
      className="entryIcon_ull2p3-o_O-passedBadge_1gkx146"
      color="secondary"
      pill={true}
      tag="span"
      title="testplan"
    >
      TP
    </Badge>
  </div>
</div>
`;
